### PR TITLE
Added new entry on Software Engineering

### LINKED ISSUE #1822: [Entry] General: Software Engineering
### Reason/inspiration (optional)

After I submitted #1821, I realized there isn't an entry on software engineering. 

If you want to have it added/assigned, feel free to assign this issue to me. Otherwise, I can request it if it's still open once I've sent a PR for #1821 

### Entry Type (select all that apply)

New Concept Entry for Existing Topic

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by YanisaHS**
### Description

Added a new entry on Software Engineering to close #1822 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/general/concepts/software-engineering/software-engineering.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  - 'Computer Science'
+Tags:
+  - 'Development'
+  - 'Servers'
+  - 'Frameworks'
+  - 'Developer Tools'
+CatalogContent:
+  - 'learn-how-to-code'
+  - 'paths/full-stack-engineer-career-path'
+  - 'paths/back-end-engineer-career-path'
```
Comment (caupolicandiaz) [inline content/general/concepts/software-engineering/software-engineering.md:16]:
```suggestion
```

Comment (caupolicandiaz) [reply content/general/concepts/software-engineering/software-engineering.md:16]: We limit the subject and content fields to two items, tags can contain as many as five

File: content/general/concepts/software-engineering/software-engineering.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
+---
+Title: 'Software Engineering'
+Description: 'Software engineering is a field of study that involves using engineering principles to design, build, test, and maintain software systems.'
+Subjects:
+  - 'Mobile Development'
```
Comment (caupolicandiaz) [inline content/general/concepts/software-engineering/software-engineering.md:5]:
```suggestion
```

File: content/general/concepts/software-engineering/software-engineering.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+## Types of Software Engineers
+
+Software engineers are commonly categorized into three types: front-end, back-end, and full-stack.
+
+### Front-End Engineers
+
+A [front-end](https://www.codecademy.com/resources/docs/general/front-end) engineer creates and manages the user-facing components of an application, such as a website. They typically work on web applications, although some front-end engineers work on desktop and mobile applications. Front-end engineers are familiar with languages such as JavaScript, [TypeScript](https://www.codecademy.com/resources/docs/typescript), [Ruby](https://www.codecademy.com/resources/docs/ruby), [JSON](https://www.codecademy.com/resources/docs/general/json), [HTML](https://www.codecademy.com/resources/docs/html), and [CSS](https://www.codecademy.com/resources/docs/css). They also use web development tools and frameworks, such as [Angular](https://www.codecademy.com/resources/docs/open-source/angular), [React](https://www.codecademy.com/resources/docs/react), and [Bootstrap](https://www.codecademy.com/resources/docs/open-source/bootstrap).
```
Comment (caupolicandiaz) [inline content/general/concepts/software-engineering/software-engineering.md:35]:
```suggestion
A [front-end](https://www.codecademy.com/resources/docs/general/front-end) engineer creates and manages the user-facing components of an application, such as a website. They typically work on web applications, although some front-end engineers work on desktop and mobile applications. Front-end engineers are familiar with programming languages such as JavaScript, [TypeScript](https://www.codecademy.com/resources/docs/typescript), [Ruby](https://www.codecademy.com/resources/docs/ruby), as well as mark-up languages [HTML](https://www.codecademy.com/resources/docs/html), and [CSS](https://www.codecademy.com/resources/docs/css). They also use web development tools and frameworks, such as [Angular](https://www.codecademy.com/resources/docs/open-source/angular), [React](https://www.codecademy.com/resources/docs/react), and [Bootstrap](https://www.codecademy.com/resources/docs/open-source/bootstrap).
```

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@YanisaHS I've added a few comments. Please update, and tag me when you've finished. Thanks

**Comment by YanisaHS** [comment]
@caupolicandiaz thank you for your feedback!! I've finished committing those changes 🥳

**Review by caupolicandiaz — APPROVED** [review]
@YanisaHS LGTM. This is ready for another review.

**Review by KTom101 — APPROVED** [review]
@YanisaHS I just added a couple of minor changes in commit [fd20d56](https://github.com/Codecademy/docs/pull/2041/commits/fd20d5647c4f7016a0f40e842ad5a35d1dbdc11f). Looks good to merge, thank you 👍

**Comment by KTom101** [comment]
@YanisaHS Entry is live: https://www.codecademy.com/resources/docs/general/software-engineering 🎉 Thanks for another solid contribution and your patience on this! 🎉

Screenshot 2023-05-12 at 6 35 47 PM

**Pull request merged**